token/program-2022/src/instruction.rs

```diff
@@ -251,6 +264,9 @@ pub enum TokenInstruction {
     /// decimals value is checked by the caller.  This may be useful when
     /// creating transactions offline or within a hardware wallet.
     ///
+    /// If the accounts contain `AccountTransferFeeMixin`s, the fee is withheld in the
+    /// destination account.
```

Comment:
I thought we landed on the fee always being withheld in the source account?
```suggestion
    /// source account.
```

Reply:
We went back and forth about it.  Destination is probably better in the end for the clearer closing behavior.  For example, if my account was recently harvested, and I haven't received any transfers since, I can do `transfer` + `close` in one instruction without the permissionless harvest.  What do you think?

Reply:
I agree destination is nicer for the normal case, I would prefer it. 

I thought you had a pretty good argument about griefing the fee collector but moving the fee around continually.  But with `HarvestFeeToMint` the situation seems better.  The fee collector script, in a loop, would:
1. Fetch all known token accounts with a zero withheld tokens 
2. Batch them into calls to `HarvestFeeToMint`
so they'd reclaim the fee eventually!

Reply:
To be clear, I don't think the whole wittheld amount should be sent over, just that on one transfer, the wittheld amount stays in the destination.  For example, if:
* transfer fee is 1%
* my account has 10 tokens withheld,
* yours has 3 tokens withheld

I send you 100 tokens. Afterwards:

* my account still has 10 tokens withheld total
* your account receives 99 tokens to your balance
* your withheld amount goes up by 1, to 4